### PR TITLE
Added support for tinyint to _parse.py

### PR DESCRIPTION
**Pull request by TimTheinAtTabs**
Using SqlAlchemy to reflect on a table having a tinyint column throws an error because tinyint type was missing from the type map. Adding a single line to the code seems to have fixed the bug.

### CODE REVIEW

**Review by susodapop — APPROVED** [review]
Thank you! We'll need to add a couple tests for this behaviour before merging.

**Comment by susodapop** [comment]
Please push an update to your commit including the DCO sign-off as described [here](https://github.com/databricks/databricks-sql-python/blob/main/CONTRIBUTING.md#sign-your-work).

File: src/databricks/sqlalchemy/_parse.py
Comment (susodapop) [inline src/databricks/sqlalchemy/_parse.py:285]: This should be  `type_overrides.TINYINT`, not `SmallInteger`, no?

**Review by susodapop — CHANGES_REQUESTED** [review]
Sorry for the chatter. I clicked the approve button prematurely. One comment for you:

Before we can merge:
1. We need to add e2e / unit tests for this. I can do this for you if you're not able to do so easily.
2. We can't merge this change without a DCO sign-off on your commit. Please amend your commit and push it to your branch.

File: src/databricks/sqlalchemy/_parse.py
Comment (TimTheinAtTabs) [reply src/databricks/sqlalchemy/_parse.py:285]: I do not know enough of the code to make the judgement call. I used my best guess to figure out the quickest solution that could work.

Comment (susodapop) [reply src/databricks/sqlalchemy/_parse.py:285]: Understood. Databricks has both `SMALLINT` and `TINYINT` types. The code you're editing has to do with column reflection of existing tables. These are mapped to SqlAlchemy types named `SmallInteger` (which is built-in to SQLAlchemy) and `TINYINT` (which is a custom type implemented in our dialect). In Python, a `TINYINT` in our dialect behaves _exactly_ the same as a `SmallInteger` (Python doesn't have any size restrictions for integers). That's why your version of the code _works for your use case_. But we should reflect the type properly back as `type_overrides.TINYINT. Otherwise SQLAlchemy will not properly reflect back the true schema of the table, which will be an issue for people using SQLAlchemy/Alembic to programmatically manage their table schemas.

**Comment by susodapop** [comment]
I've pushed a fix that properly tests this behaviour. Can you update your commit with the DCO sign-off?

File: src/databricks/sqlalchemy/test_local/test_types.py
Comment (susodapop) [inline src/databricks/sqlalchemy/test_local/test_types.py:114]: I found during testing that this test case would always pass because prefixing an assertion with `return` always evaluates as a pass. When I removed the `return`, the test failed for TINYINT types (should have been caught during initial development). This test now passes.

**Review by benc-db — APPROVED** [review]

**Pull request merged**
